std/container/slist.d

```diff
@@ -359,39 +401,15 @@ Returns: The number of elements inserted
 Complexity: $(BIGOH m), where $(D m) is the length of $(D stuff)
      */
     size_t insertFront(Stuff)(Stuff stuff)
-    if (isInputRange!Stuff && isImplicitlyConvertible!(ElementType!Stuff, T))
```

Comment:
Probably should combine the two constraints with an `||` rather than getting rid of them

Reply:
Would certainly be correct. I chose to do it like it already was on `insertAfter`. Willing to change.

Reply:
Hmm, IIRC there were some issues with `isImplicitlyConvertible` and @jmdavis has been working on slowly fixing Phobos for this.
The problem essentially boils down to static arrays been implicitly convertible to strings:

```
static assert( isImplicitlyConvertible!(immutable(char)[4], string));
```

which means that the string slice could point to memory that is no longer on the heap...

Reply:
> I chose to do it like it already was on insertAfter. Willing to change.

I think it would be better to keep the constraints and use `isInputRange!Stuff || isImplicitlyConvertible!(Stuff, T)` as then the user will see an error message in his code and not in Phobos code.
Probably a good idea to do the same for `insertAfter` too.